util/file_reader_writer.cc

```diff
@@ -512,19 +513,24 @@ class ReadaheadRandomAccessFile : public RandomAccessFile {
         return Status::OK();
       }
     }
+    size_t aligned_offset = TruncateToPageBoundary(alignment_, offset + copied);
     Slice readahead_result;
-    Status s = file_->Read(offset + copied, readahead_size_, &readahead_result,
-      buffer_.get());
+    Status s = file_->Read(aligned_offset, readahead_size_ + alignment_,
+                           &readahead_result, buffer_.get());
```

Comment:
I mean do we guarantee the memory pointed by buffer_.get() is page aligned? In other places where we need to do aligned read, we align the pointer to the page boundary. For example see what we do in class AlignedBuffer: https://github.com/facebook/rocksdb/blob/master/util/aligned_buffer.h. Maybe we can just use AlignedBuffer instead.

Sorry I didn't realize this problem when I created the task. If it is too complicated to make the change, you can leave the buffer_.get(), and we can follow up the change later.